src/rewriter/README.md

```diff
@@ -22,35 +21,20 @@ This document describe the syntax of the RARE language, which stands for
 
 ### Matching
 
-The basic form of a RARE rewrite rule uses `define-rule`. The two
-expressions in the body of `define-rule` indicate the *match* and *target*
-of the rewrites.
+The basic form of a RARE rewrite rule uses `define-rule`. The two expressions in
+the body of `define-rule` indicate the *match* and *target* of the rewrites.
 
 ``` lisp
 (define-rule substr-empty
 	((m Int) (n Int))
 	(str.substr "" m n)
 	"")
 ```
-In this case, the match is `(str.substr "" m n)` and the target is `""`. For example, when the rule is applied to
-`(str.substr "" 3 5)`, `m,n` are instantiated to `3,5`, respectively, and the expression is rewritten as `""`.
+In this case, the match is `(str.substr "" m n)` and the target is `""`. For example, when the rule is applied to `(str.substr "" 3 5)`, `m,n` are instantiated to `3,5`, respectively, and the expression is rewritten as `""`.
 
 Match is purely syntactic. It matches *syntactically identical* terms. For
 example `(= (str.++ x1 x2) x2)`matches `a ++ b = b` but not `a ++ b = c`.
 
-### Fixed-Point Rules
-
-The `*` in `define-rule*` indicates that the rule shall be executed until it can
-no longer apply. In this case the user can supply a rewrite context to support
-the continuation.
-
-``` lisp
-(define-rule* str-len-concat-rec ((s1 String) (s2 String) (s3 String :list))
-	(str.len (str.++ s1 s2 s3))
-	(str.len (str.++ s2 s3))
-	(+ (str.len s1) _))
-```
-This rule specifies that the rewrite of `(str.len (str.++ s1 s2 ...))` is the sum of `(str.len s1)` and the result of applying the rule to the term `(str.len (str.++ s2 ...))`
 
 ### Conditional Rules
```

Comment:
You mention that "In define-cond-rule, an additional expression immediately after the parameter and definition list restricts the rule to certain cases." but have not yet explained or even mentioned definition lists. It might be worth moving that part up before describing conditions.

Reply:
done